### ISSUE TITLE
Injected classes in jax-rs resource results in unexpected body parameter

### ISSUE THREAD

**Issue by foxylion** [swagger-api/swagger-core#1176]
Hi,
I hit a problem which seems to be, accordingly to ron of the IRC channel, a bug.

Given is a resource class as following, which is using dependency injection (in my case with guice):
```java
@Path("/my-account")
@Produces(MediaType.APPLICATION_JSON)
@Api(tags = { "Users" })
public class MyAccountResource {

    private final MyAccountResourceImpl delegate;

    @javax.inject.Inject
    public MyAccountResource(MyAccountResourceImpl delegate) {
        this.delegate = delegate;
    }

    @GET
    @ApiOperation(value = "User details of the authenticated user", notes = "Redirects to the actual user resource", code = 303)
    public Response getMyAccountDetail() {
        return delegate.getMyAccountDetail();
    }
}
```

A snippet of the generated json exposes a `body` parameter for the get request with scheme `MyAccountResourceImpl` which is not expected.

```json
    "/my-account" : {
      "get" : {
        "tags" : [ "Users" ],
        "summary" : "User details of the authenticated user",
        "description" : "Redirects to the actual user resource",
        "operationId" : "getMyAccountDetail",
        "produces" : [ "application/json" ],
        "parameters" : [ {
          "in" : "body",
          "name" : "body",
          "required" : false,
          "schema" : {
            "$ref" : "#/definitions/MyAccountResourceImpl"
          }
        } ],
        "responses" : {
          "default" : {
            "description" : "successful operation"
          }
        }
      }
    },
```

Good news for everyone who is using guice as the dependency injection framework - just use the `@com.google.inject.Inject` annotation instead of `@javax.inject.Inject` as a workaround.

**Comment by iushankin** [comment]
Will be fixed in #1191

**Comment by webron** [comment]
Fixed in develop_2.0.
